gino/transaction.py

```diff
@@ -166,16 +166,16 @@ def __await__(self):
         return self
 
     async def __aexit__(self, ex_type, ex, ex_tb):
-        try:
-            is_break = ex_type is _Break
-            if is_break and ex.commit:
-                ex_type = None
-            if ex_type is None:
+        is_break = ex_type is _Break
+        if is_break and ex.commit:
+            ex_type = None
+        if ex_type is None:
+            try:
                 await self._tx.commit()
-            else:
+            except Exception:
```

Comment:
we only expect exceptions when committing, so I moved the try..except block here

Reply:
Yeah this makes sense.